### DISCUSSION TITLE
how to let `mini.comment` comment c++ code with `//...` instead of `/*...*/`

### DISCUSSION THREAD

**Discussion by lainegates** [nvim-mini/mini.nvim#669] [Q&A] [answered]
Is there a config parameter that let `mini.comment` comment c++ code with `//...` instead of `/*...*/`

**Comment by echasnovski** [comment, marked as answer]
'mini.comment' infers comment structure [based on 'commentstring' option](https://github.com/echasnovski/mini.nvim/blob/1cd8d7e8318071eddc536815f5cafe50b012ac29/doc/mini-comment.txt#L11-L14).

My suggestion would be to create a file 'after/ftplugin/cpp.lua' and put `vim.bo.commentstring = '// %s'` line there.

**Reply by lainegates** [reply]
@echasnovski Thanks!
 Could I make this setting effective only for C++ code? How to do it?

**Reply by echasnovski** [reply]
> Could I make this setting effective only for C++ code? How to do it?

Yes:

> My suggestion would be to create a file 'after/ftplugin/cpp.lua' and put vim.bo.commentstring = '// %s' line there.

**Reply by lainegates** [reply]
I'm not very familiar with the configuration of nvim. I want to confirm, do you mean to create a directory mini.comment/after/ftplugin/cpp.lua? (which means code of "mini.comment" will be modified.)
Or should it be in ~/.config/nvim/lua/after/ftplugin/cpp.lua?"

**Reply by echasnovski** [reply]
The '~/.config/nvim/after/ftplugin/cpp.lua' (*no* 'lua' subdirectory). Files inside 'ftplugin' directory (either directly or inside 'after' directory) are considered to have "filetype plugins" - files which are sourced when 'filetype' option is set to the value same as file's name.
You can read more about plugin structure typing `:h 'runtime'` (followed by `<CR>`, a.k.a. `<Enter>`).

**Reply by lainegates** [reply]
@echasnovski Thanks! It works, I will read more about plugins.
